### PR TITLE
Callout positioning: update useMaxHeight hook for top edge alignment; account for scroll resizing when choosing callout position.

### LINKED ISSUE #28954: [Bug]: Dropdown callout renders to the right when list content exceeds vertical space
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 16.38 GB / 31.59 GB
  Browsers:
    Edge: Spartan (44.22621.2134.0), Chromium (115.0.1901.203)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/nipope1/pen/PoXqPbQ

## Actual Behavior
Callout renders to the right of dropdown when list content exceeds vertical space. When the width of the dropdown is equal to the width of the screen, the callout takes up the entire screen when opened (2nd example in attached Codepen).

## Expected Behavior
Callout should render below or (if necessary) above dropdown with a scrollbar if content exceeds vertical space

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

Azure Deployment Environments (Azure Portal extension)

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #29319: [Bug]: Callout positioning does not account for scrollable content
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) ia32 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 20.71 GB / 31.95 GB
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (116.0.1938.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/nipope1/pen/jOXBOvO

## Actual Behavior
When `directionalHint = top...` and `directionalHintFixed = true`, and callout content exceeds the remaining height of the viewport, callout vertically fills the entire viewport.

(I put together this pen to confirm that the issue was specifically with callout, and not with dropdown: https://codepen.io/nipope1/pen/MWZGxzw)

## Expected Behavior
Callout should pin to the top of its target and fill the remaining space to the top of the viewport.

CodePen that shows working behavior in Fluent v7: https://codepen.io/nipope1/pen/poqVYaM

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

Azure Deployment Environments (Azure Portal extension)

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by nipope**
## Previous Behavior

1. Issue #29319 : callout does not align to the top of the target when targetEdge is top and content exceeds available space
2. Issue #28954 : When flipping sides to find the best fit, callout does not account for scroll resizing and will never choose top / bottom edges when content exceeds page vertical space. 
   - (Ex. directionalHint is bottomAutoEdge (directionalHintFixed = false) and callout content exceeds the height of the screen - the callout will always reposition to the right of the target, even though there is sufficient space below to make the callout scrollable)

## New Behavior

1. When target edge is top, callout will at maximum fill the space between the top of the screen and the top of the target, regardless of content size
2. When callout content exceeds available space below / above the target, the callout will still consider these edges as candidates for the final callout position if there is sufficient space to scroll resize.
   - (Ex. directionalHint is bottomAutoEdge (directionalHintFixed = false) and callout content exceeds the height of the screen - the callout will now be positioned at the bottom edge of the target)

## Related Issue(s)

- Fixes #29319
- Fixes #28954

### CODE REVIEW

File: packages/react/src/utilities/positioning/positioning.ts
Comment (nipope) [inline packages/react/src/utilities/positioning/positioning.ts:226]: I chose 100 arbitrarily here as the amount of space that we can reasonably scroll-resize down to (and I don't think it's the right value - this is too small of a space). Would it make sense to allow consumers to define this in props, and have a default value?

**Comment by nipope** [comment]
A question on overall approach (regarding the scroll-resizing part of this PR) - should there be a prop (something like `preferScrollResizing`) that allows consumers to opt into this change? Or is it an ok behavior to always consider scroll-resizing preferable to flipping to a new edge?

File: change/@fluentui-react-c8535f3e-253b-40b8-92e7-bb6c8e552afe.json
Comment (fabricteam) [inline change/@fluentui-react-c8535f3e-253b-40b8-92e7-bb6c8e552afe.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29766&commitId=d83e1a0a53d2d24590abd6b4b0f89bf5ec363760&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Rendering callout attached to a rectangle.default.chromium.png|1604386| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Multiple - Dark Mode.hover.chromium.png|4360| Changed|
|react-charting-AreaChart.Multiple.hover.chromium.png|4169| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/BaseButton-mount_base.html">639</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/BaseButton-mount.html">633</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Breadcrumb-mount_base.html">1686</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Breadcrumb-mount.html">1721</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Checkbox-mount_base.html">1695</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Checkbox-mount.html">1664</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/CheckboxBase-mount_base.html">1434</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/CheckboxBase-mount.html">1467</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ChoiceGroup-mount_base.html">2965</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ChoiceGroup-mount.html">2893</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ComboBox-mount_base.html">666</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ComboBox-mount.html">676</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/CommandBar-mount_base.html">6146</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/CommandBar-mount.html">6094</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ContextualMenu-mount_base.html">12302</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ContextualMenu-mount.html">12279</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DefaultButton-mount_base.html">732</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DefaultButton-mount.html">767</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DetailsRow-mount_base.html">2177</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DetailsRow-mount.html">2171</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DetailsRowFast-mount_base.html">2215</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DetailsRowFast-mount.html">2173</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DetailsRowNoStyles-mount_base.html">2031</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DetailsRowNoStyles-mount.html">2041</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Dialog-mount_base.html">2639</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Dialog-mount.html">2608</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DocumentCardTitle-mount_base.html">232</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/DocumentCardTitle-mount.html">229</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Dropdown-mount_base.html">1965</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Dropdown-mount.html">2009</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/FocusTrapZone-mount_base.html">1121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/FocusTrapZone-mount.html">1128</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/FocusZone-mount_base.html">1078</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/FocusZone-mount.html">1069</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedList-mount_base.html">41673</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedList-mount.html">41647</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedList-virtual-rerender_base.html">19864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedList-virtual-rerender.html">20022</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">50847</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedList-virtual-rerender-with-unmount.html">50559</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedListV2-mount_base.html">214</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedListV2-mount.html">233</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedListV2-virtual-rerender_base.html">213</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedListV2-virtual-rerender.html">217</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">221</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">220</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/IconButton-mount_base.html">1065</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/IconButton-mount.html">1074</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Label-mount_base.html">333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Label-mount.html">334</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Layer-mount_base.html">2698</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Layer-mount.html">2738</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Link-mount_base.html">379</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Link-mount.html">391</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/MenuButton-mount_base.html">966</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/MenuButton-mount.html">946</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/MessageBar-mount_base.html">21309</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/MessageBar-mount.html">21273</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Nav-mount_base.html">1941</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Nav-mount.html">1959</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/OverflowSet-mount_base.html">759</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/OverflowSet-mount.html">767</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Panel-mount_base.html">1772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Panel-mount.html">1749</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Persona-mount_base.html">711</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Persona-mount.html">750</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Pivot-mount_base.html">870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Pivot-mount.html">862</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/PrimaryButton-mount_base.html">839</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/PrimaryButton-mount.html">847</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Rating-mount_base.html">4532</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Rating-mount.html">4563</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SearchBox-mount_base.html">898</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SearchBox-mount.html">928</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Shimmer-mount_base.html">1888</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Shimmer-mount.html">1878</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Slider-mount_base.html">1308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Slider-mount.html">1316</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SpinButton-mount_base.html">2852</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SpinButton-mount.html">2818</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Spinner-mount_base.html">374</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Spinner-mount.html">378</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SplitButton-mount_base.html">1839</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SplitButton-mount.html">1839</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Stack-mount_base.html">414</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Stack-mount.html">418</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/StackWithIntrinsicChildren-mount_base.html">839</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/StackWithIntrinsicChildren-mount.html">872</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/StackWithTextChildren-mount_base.html">2625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/StackWithTextChildren-mount.html">2604</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SwatchColorPicker-mount_base.html">6081</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/SwatchColorPicker-mount.html">6109</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/TagPicker-mount_base.html">1423</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/TagPicker-mount.html">1545</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Text-mount_base.html">371</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Text-mount.html">372</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/TextField-mount_base.html">913</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/TextField-mount.html">944</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ThemeProvider-mount_base.html">832</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ThemeProvider-mount.html">838</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ThemeProvider-virtual-rerender_base.html">576</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ThemeProvider-virtual-rerender.html">577</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1255</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1270</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Toggle-mount_base.html">623</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/Toggle-mount.html">607</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/buttonNative-mount_base.html">186</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29766/perf-test/buttonNative-mount.html">201</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/utilities/positioning/positioning.ts
Comment (nipope) [reply packages/react/src/utilities/positioning/positioning.ts]: I ended up adding a `minimumScrollResizeHeight` prop to Callout

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by spmonahan** [comment]
> A question on overall approach (regarding the scroll-resizing part of this PR) - should there be a prop (something like `preferScrollResizing`) that allows consumers to opt into this change? Or is it an ok behavior to always consider scroll-resizing preferable to flipping to a new edge?

I think an opt in prop is the right approach. The existing behavior, for better or worse, is the default expected behavior at this point so we should leave that as-is.

**Comment by nipope** [comment]
> > A question on overall approach (regarding the scroll-resizing part of this PR) - should there be a prop (something like `preferScrollResizing`) that allows consumers to opt into this change? Or is it an ok behavior to always consider scroll-resizing preferable to flipping to a new edge?
> 
> I think an opt in prop is the right approach. The existing behavior, for better or worse, is the default expected behavior at this point so we should leave that as-is.

Sounds good! Just pushed more changes, including an opt-in prop for this behavior.

**Review by sopranopillow — APPROVED** [review]
Looks great! once the comments are addressed, we can go ahead and merge it :)

File: packages/react/src/components/Callout/CalloutContent.base.tsx
Comment (nipope) [reply packages/react/src/components/Callout/CalloutContent.base.tsx:23]: Should these be added to `utilities/positioning/index.ts` and imported from there?

**Review by sopranopillow — APPROVED** [review]

**Comment by micahgodbolt** [comment]
looking to get one more Fluent UI team approval before force merging this in.

**Comment by spmonahan** [comment]
> looking to get one more Fluent UI team approval before force merging this in.

Only needs force merge to override the bundle size analysis right?

File: packages/react/src/components/Callout/CalloutContent.base.tsx
Comment (nipope) [reply packages/react/src/components/Callout/CalloutContent.base.tsx:210]: Sounds good to me! Is it possible that the styles would change while popupRef.current points to the same object?

Comment (spmonahan) [reply packages/react/src/components/Callout/CalloutContent.base.tsx]: > Is it possible that the styles would change while popupRef.current points to the same object?

Yes, but the [returned object is live](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle#return_value) so it will be updated too 🙂

Comment (nipope) [reply packages/react/src/components/Callout/CalloutContent.base.tsx]: Oh makes sense! Thank you for the info! :)

**Review by spmonahan — APPROVED** [review]

**Pull request merged**
